ttg/ttg/parsec/ttg.h

```diff
@@ -554,40 +532,54 @@ namespace ttg_parsec {
     }
 
     inline void release_data_copy(ttg_data_copy_t *copy) {
-      if (nullptr != copy->device_private) {
-        if (copy->readers > 0) {
-          int32_t readers = parsec_atomic_fetch_dec_int32(&copy->readers);
-          if (1 == readers) {
-            copy->delete_fn(copy->device_private);
-            copy->device_private = NULL;
-          }
-        }
-      }
       if (NULL != copy->push_task) {
-        /* Release the task if it was deferrred */
+        /* Release the the deferred task.
+         * The copy was mutable and will be mutated by the released task,
+         * so simply transfer ownership.
+         */
         parsec_task_t *push_task = copy->push_task;
-        if (parsec_atomic_cas_ptr(&copy->push_task, push_task, nullptr)) {
-          parsec_ttg_task_base_t *deferred_op = (parsec_ttg_task_base_t *)copy->push_task;
-          assert(deferred_op->deferred_release);
-          deferred_op->deferred_release(deferred_op->tt_ptr, deferred_op);
+        copy->push_task = nullptr;
+        parsec_ttg_task_base_t *deferred_op = (parsec_ttg_task_base_t *)push_task;
+        assert(deferred_op->deferred_release);
+        deferred_op->deferred_release(deferred_op->tt_ptr, deferred_op);
+      } else {
+        if (copy->readers < 0) {
```

Comment:
How it is possible to have a negative number of readers ? Especially when the current task is supposed to hold at least one of the reader positions.

Reply:
A negative value in readers marks the copy as being mutable. So the copy either belongs `readers > 0` (a bunch of readers) or `readers == INT32_MIN` (a single task mutating the copy).

Reply:
Fix to check for `INT32_MIN` instead of smaller-than.